packages/react-instantsearch-dom/src/lib/infiniteHitsCache/__tests__/sessionStorage.js

```diff
@@ -0,0 +1,44 @@
+import getInfiniteHitsSessionStorageCache from '../sessionStorage';
+
+describe('getInfiniteHitsSessionStorageCache', () => {
+  let sessionStorageBackup;
+  const store = {};
+  beforeAll(() => {
+    sessionStorageBackup = window.sessionStorage;
+  });
+  beforeEach(() => {
+    window.sessionStorage = {
+      getItem(key) {
+        return store[key];
+      },
+      setItem(key, value) {
+        store[key] = value;
+      },
+    };
+  });
+  afterAll(() => {
+    window.sessionStorage = sessionStorageBackup;
+  });
+
+  it('returns null initially', () => {
+    const cache = getInfiniteHitsSessionStorageCache();
+    expect(cache.read({})).toBeNull();
+  });
+
+  it('returns what it was assigned before', () => {
+    const cache = getInfiniteHitsSessionStorageCache();
+    const state = { q: 'hello' };
+    const hits = { 1: ['a', 'b', 'c'] };
+    cache.write({ state, hits });
+    expect(cache.read({ state })).toEqual(hits);
+  });
+
+  it('returns null if the state is different', () => {
+    const cache = getInfiniteHitsSessionStorageCache();
+    const state = { q: 'hello' };
+    const newState = { q: 'world' };
+    const hits = { 1: ['a', 'b', 'c'] };
+    cache.write({ state, hits });
+    expect(cache.read({ state: newState })).toBeNull();
+  });
```

Comment:
I'd like to see a test where the cache (mock) throws an error on creation or on access, do you think that makes sense?

Reply:
I was assuming user passes a correct cache object with correct methods (read, write) with exception handling. That's why I try - catched inside `sessionStorage`. Do you want to handle exceptions on the connector level?

Reply:
It's the cache which catches the error, right?

Reply:
Yes, but I'm not sure if I understand your first comment.

> where the cache (mock) throws an error on creation or on access

```jsx
<InfiniteHits
  hitComponent={Product}
  cache={createInfiniteHitsSessionStorageCache()}
/>
```

On creation -> `createInfiniteHitsSessionStorageCache()` call?


```js
# connectInfiniteHits

this._cachedHits = cache.read({ state: searchState }) || {};
```
On access -> `.read()` or `.write()` ?

Reply:
What I meant is to write some tests for cases where the underlying session storage fails at different moments (creation, read, invalid json) with this cache wrapper

Reply:
ah you meant sessionStorage. got it.

Reply:
wdyt?
https://github.com/algolia/react-instantsearch/pull/2921/commits/89aedc6a4ca5a2e4442a8746a18fd1ee737e6435